File: LeetCode/Easy/3731-find-missing-elements/3731-find-missing-elements.cpp
class Solution {
public:
    vector<int> findMissingElements(vector<int>& nums) {
        if (nums.empty()) return {};
        int mn = nums[0], mx = nums[0];
        for (int x : nums) {
            mn = min(mn, x);
            mx = max(mx, x);
        }
        vector<bool> has(mx - mn + 1, false);
        for (int x : nums) {
            has[x - mn] = true;
        }
        vector<int> ans;
        for (int i = 0; i < has.size(); i++) {
            if (!has[i]) {
                ans.push_back(mn + i);
            }
        }
        return ans;
    }
};

File: LeetCode/Easy/3731-find-missing-elements/3731-find-missing-elements.py
class Solution:
    def findMissingElements(self, nums: list[int]) -> list[int]:
        if not nums:
            return []
        mn, mx = min(nums), max(nums)
        has = set(nums)
        return [x for x in range(mn + 1, mx) if x not in has]
